Remove p8platform

phunkyfish opened the pull request:
v6.0.2
- Remove p8-platform dependency
- Remove p8-platform OS includes
- Use chrono to get millis time
- Use std::mutex
- Use kodi StringUtils
- Remove SAFE_DELETE
- Remove p8-platform time utils
- Use chrono for timing
- Use std::thread
- Include windows.h for LARGE_INTEGER, MAX_PATH and GetTempAthA support
- add temporary workaround about attribute hidden

phunkyfish committed:
Use std::thread

phunkyfish committed:
Use chrono for timing

phunkyfish committed:
Remove p8-platform time utils

phunkyfish committed:
Remove SAFE_DELETE

phunkyfish committed:
Use kodi StringUtils

phunkyfish committed:
Use std::mutex

phunkyfish committed:
Use chrono to get millis time

phunkyfish committed:
Remove p8-platform OS includes

phunkyfish committed:
Remove p8-platform dependency

phunkyfish committed:
changelog and version v6.0.2

phunkyfish commented:
Quite a lot of change here so will need a proper review.

Can you also runtime test @AlwinEsch ?

AlwinEsch commented on src/KeepAliveThread.cpp:
```diff
@@ -23,22 +22,45 @@ CKeepAliveThread::CKeepAliveThread(cPVRClientArgusTV& instance) : m_instance(ins
   {
     int retval = m_instance.GetRPC().KeepLiveStreamAlive();
     kodi::Log(ADDON_LOG_DEBUG, "CKeepAliveThread:: KeepLiveStreamAlive returned %i", (int)retval);
-    // The new P8PLATFORM:: thread library has a problem with stopping a thread that is doing a long sleep
```
Think this also present by way new here and inside Kodi it was done by:
```cpp
  void Sleep(uint32_t milliseconds)
  {
    if (milliseconds > 10 && IsCurrentThread())
    {
      std::unique_lock<std::recursive_mutex> lock(m_threadMutex);
      m_stopEvent.wait_for(lock, std::chrono::milliseconds(milliseconds));
    }
    else
    {
      std::this_thread::sleep_for(std::chrono::milliseconds(milliseconds));
    }
  }
```
To stop direct after stopping signal is come

AlwinEsch commented on src/lib/tsreader/MultiFileReader.cpp:
```diff
@@ -43,8 +41,9 @@
+#include <kodi/tools/EndTime.h>
 
-using namespace P8PLATFORM;
+//using namespace P8PLATFORM;
```
This line can be then deleted.

AlwinEsch commented on src/lib/tsreader/MultiFileReader.cpp:
```diff
@@ -398,7 +397,8 @@ long MultiFileReader::RefreshTSBufferFile()
       m_TSBufferFile.OpenFile();
-      usleep(5000);
+      std::this_thread::sleep_for(std::chrono::milliseconds(5));
+
```
No new line needed here.

AlwinEsch commented on src/tools.cpp:
```diff
@@ -12,9 +12,8 @@
-#include <p8-platform/util/timeutils.h>
 
-using namespace P8PLATFORM;
+#include <chrono>
```
As also in header not required here.

AlwinEsch commented on src/lib/tsreader/platform.h:
```diff
@@ -0,0 +1,147 @@
+  return (i);
+};
+
+#endif
```
Maybe this not needed, has looked to change without here https://github.com/AlwinEsch/pvr.argustv/commit/cb5b9c56d724de0c252c13e7695eafba96a6613a

Need to check compile and runtime also on Windows and I'm report later.

phunkyfish replied:
Without would be way better.

AlwinEsch reviewed (commented):
Looks mostly OK, thing only the platform.h not really needed.

Looking currently to become my backend working again.
The connect to there was already OK, only has it not worked with stream (backend self related error) and make it now working again.

AlwinEsch committed:
add temporary workaround about attribute hidden

This becomes in future added to dev-kit system, only for to have warnings away.

phunkyfish committed:
Include windows.h for LARGE_INTEGER, MAX_PATH and GetTempAthA support

phunkyfish committed:
Refactoring/cleanup

AlwinEsch reviewed (approved):
Is good to come in, about the use of inputstream intance within, I must look more detailed how the best to bring (without playback stop :smirk:).

But was good to see that possible :smile:
